### PR TITLE
Add spellchecking to pre-commit hooks and linting workflow

### PR DESCRIPTION
**Pull request by eliykat**
## Objective

Add spellchecking, using [cspell](https://cspell.org/). Because we all make typos.

## Code changes
* install cspell as a dev dependency
* add configuration in `cspell.yaml`, largely following [the documentation](https://cspell.org/docs/dictionaries-custom/)
* add custom word list
* add npm script, add to lint-staged, and add to Github linting workflow
* the good part: fix typos 😁

### CODE REVIEW

**Comment by eliykat** [comment]
This is currently excluding `.mdx` files, but I'm having issues with glob pattern matching in npm scripts (probably because of [this issue](https://medium.com/@jakubsynowiec/you-should-always-quote-your-globs-in-npm-scripts-621887a2a784)) and regex generally. Feedback welcome in the meantime.

File: docs/contributing/index.md [hunk 1]
```diff
@@ -16,11 +16,11 @@ Here is how you can get involved:
...
-- **Write code for a new feature:** Create a new
-  [Github Discussions](https://github.com/orgs/bitwarden/discussions) post outlining your proposal.
-  Include a description of your proposed contribution, screenshots, and links to any relevant
-  [feature requests](https://community.bitwarden.com/c/feature-requests/5/none). This helps get
-  feedback from the community and Bitwarden team members before you start writing code
+- **Write code for a new feature:** Make a new post in the
+  [Github Contributions category](https://community.bitwarden.com/c/github-contributions/) of the
+  Community Forums. Include a description of your proposed contribution, screenshots, and links to
+  any relevant feature requests. This helps get feedback from the community and Bitwarden team
+  members before you start writing code
```
Comment (Hinton) [inline docs/contributing/index.md:23]: We seem to have shuffled around text here.

File: docs/architecture/deep-dives/autofill/shadow-dom.md [hunk 2]
```diff
@@ -45,7 +45,8 @@ If the node is a shadow root, we try to recursively descend into it using the br
 - **Firefox: **
   [Element.openOrClosedShadowRoot](https://developer.mozilla.org/en-US/docs/Web/API/Element/openOrClosedShadowRoot) -
   ignores the `mode` property and works for open and closed shadow roots
-- **Safari and others: **
+- **Safari and others:**
+
```
Comment (Hinton) [inline docs/architecture/deep-dives/autofill/shadow-dom.md:49]: Was this likebreak intentional? The other list items does not have a linebreak here.

**Comment by withinfocus** [comment]
I happened to come upon this, and I have an online linter proposed in our template repository at https://github.com/bitwarden/template/pull/2. I actually disabled cspell based on past experience where it's really unwieldy to maintain. You can see some alternatives that the Mega-Linter uses at https://megalinter.io/latest/descriptors/spell/.

**Comment by eliykat** [comment]
Thanks @withinfocus. I see that MegaLinter can run locally, so I think we finalize your PR (which enables it in the pipeline) and then we can add the local npm script & pre-commit hook - which I think are valuable for near-immediate feedback. (Whereas I see CI checks as the ultimate enforcer but not very convenient for devs.) Do you agree?

**Comment by withinfocus** [comment]
CI vs. local linting is definitely the most contentious part of that proposed change. I haven't personally executed it locally but will give it a try to see its performance -- my concern is its heaviness and speed despite its power.

Separately though it may be valuable to avoid cspell.

File: docs/contributing/index.md (hunk 1 above)
Comment (eliykat) [reply docs/contributing/index.md]: I fixed the link and updated the wording slightly, which caused Prettier to move things around I think.

File: docs/architecture/deep-dives/autofill/shadow-dom.md (hunk 2 above)
Comment (eliykat) [reply docs/architecture/deep-dives/autofill/shadow-dom.md]: Fixed

**Comment by eliykat** [comment]
Typo fixes have been moved to a separate PR, #129. I haven't had a lot of time to progress this one but I'll come back to it soon.

**Review by Hinton — COMMENTED** [review]
Let's fix the linting errors and this should be good.

**Pull request merged**